qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/create/offer/CustomerSelectionView.groovy

```diff
@@ -55,10 +62,11 @@ class CustomerSelectionView extends VerticalLayout{
         this.foundCustomerList = viewModel.foundCustomers
 
         initLayout()
-        setupDataProvider()
+        def customerDataProvider = setupCustomerDataProvider()
```

Comment:
I would bundle the data provider and filter setup in the grid generation since it is only useful information in this context.

Reply:
SRP

Reply:
I see your point with the SRP and I agree that it might be a good idea to have separate methods. Those methods should however be called from within the grid generation method since the data provider and the column filters are tightly coupled to exactly that grid.